src/VisualStudio/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Execution;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.ServiceHub.Client;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Remote;
+using Roslyn.Utilities;
+using StreamJsonRpc;
+
+namespace Microsoft.VisualStudio.LanguageServices.Remote
+{
+    internal partial class ServiceHubRemoteHostClient : RemoteHostClient
+    {
+        private readonly HubClient _hubClient;
+        private readonly Stream _stream;
+        private readonly JsonRpc _rpc;
+
+        public static async Task<ServiceHubRemoteHostClient> CreateAsync(Workspace workspace, CancellationToken cancellationToken)
+        {
+            var primary = new HubClient("Primary");
+            var remoteHostStream = await primary.RequestServiceAsync(WellKnownServiceHubServices.RemoteHostService, cancellationToken).ConfigureAwait(false);
+
+            var instance = new ServiceHubRemoteHostClient(workspace, primary, remoteHostStream);
+
+            // make sure connection is done right
+            var current = $"VS ({Process.GetCurrentProcess().Id})";
+            var host = await instance._rpc.InvokeAsync<string>(WellKnownServiceHubServices.RemoteHostService_Connect, current).ConfigureAwait(false);
+
+            // TODO: change this to non fatal watson and make VS to use inproc implementation
+            Contract.ThrowIfFalse(host == current.ToString());
```

Comment:
You can change this to be a non-fatal -watson.  But we shoudl not be just arbitrarily switching over to being in-proc.  We need to discuss as a team how we want this to work as such decisions have extremely large maintenance and testing costs.